src/test/java/org/jenkinsci/plugins/plaincredentials/SecretBytesTest.java

```diff
@@ -0,0 +1,190 @@
+import java.nio.charset.Charset;
+import java.util.List;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.jenkinsci.plugins.plaincredentials.impl.FileCredentialsImpl;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+import static hudson.cli.CLICommandInvoker.Matcher.succeededSilently;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+import static org.junit.Assume.*;
+
+public class SecretBytesTest {
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    /**
+     * Verifies that {@link SecretBytes} will treat a Base64 encoded plain text content as the content to be encrypted
+     * with the instance's secret key which gets applied when the {@link FileCredentialsImpl} is written to disk.
+     * @throws Exception if things go wrong.
+     */
+    @Test
+    @LocalData
+    public void loadUnencrypted() throws Exception {
+        // these are the magic strings
+        assumeThat(Base64.encodeBase64("This is Base64 encoded plain text\n".getBytes("UTF-8")), is(
+                "VGhpcyBpcyBCYXNlNjQgZW5jb2RlZCBwbGFpbiB0ZXh0Cg==".getBytes("US-ASCII")));
+
+        // first check that the file on disk contains the unencrypted text
+        assertThat(FileUtils.readFileToString(new File(r.jenkins.getRootDir(), "credentials.xml")),
+                containsString("VGhpcyBpcyBCYXNlNjQgZW5jb2RlZCBwbGFpbiB0ZXh0Cg=="));
+
+        // get the credential instance under test
+        FileCredentials c = CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(
+                        FileCredentials.class,
+                        r.jenkins,
+                        ACL.SYSTEM,
+                        (List<DomainRequirement>) null
+                ),
+                CredentialsMatchers.withId("secret-file")
+        );
+
+        // we have the credential instance we think we should have
+        assertThat(c, notNullValue());
+        assertThat(c.getFileName(), is("secret.txt"));
+        assertThat(c.getDescription(), is("a line"));
+
+        // now check that the content has been read correctly
+        assertThat(IOUtils.toString(c.getContent(), "UTF-8"), is("This is Base64 encoded plain text\n"));
+
+        // now when we re-save the credentials this should encrypt with the instance's secret key
+        SystemCredentialsProvider.getInstance().save();
+        assertThat(FileUtils.readFileToString(new File(r.jenkins.getRootDir(), "credentials.xml")),
+                not(containsString("VGhpcyBpcyBCYXNlNjQgZW5jb2RlZCBwbGFpbiB0ZXh0Cg==")));
+    }
+
+    /**
+     * Verifies that {@link SecretBytes} will treat a Base64 encoded plain text content as the content to be encrypted
+     * with the instance's secret key which gets applied when the {@link FileCredentialsImpl} is written to disk.
```

Comment:
Same Javadoc as `loadUnencrypted` yet presumably this is testing something different?